Gifski/SavePanelAccessoryViewController.swift

```diff
@@ -1,85 +1,280 @@
+                alert.informativeText = "Please provide a valid value between \(Int(metadata.dimensions.width * CGFloat(scaleXMinDoubleValue))) × \(Int(metadata.dimensions.height * CGFloat(scaleYMinDoubleValue))) px to \(Int(fileDimensions.width)) × \(Int(fileDimensions.height)) px."
+            }
+            alert.addButton(withTitle: "Discard Change")
+            alert.runModal()
+            updateTextFieldsFromPopup(asPercentage: percentageMode)
+            return false
+        } else {
+            return true
+        }
+    }
+
+    private func configureScaleSettings(inputDimensions dimensions: CGSize) {
+        for divisor in 1..<6 {
+            let divisorFloat = CGFloat(integerLiteral: divisor)
+            let dimensionRatio = Float(1 / divisorFloat)
+            dimensionRatios.append(dimensionRatio)
+            var percentageString: String = "Original"
+            if divisor != 1 {
+                percentageString = "\(Int(round(dimensionRatio * 100.0)))%"
+            }
+            predefinedSizesDropdown.addItem(withTitle: "⭐ \(Int(dimensions.width / divisorFloat)) × \(Int(dimensions.height / divisorFloat)) (\(percentageString))")
+        }
+        predefinedSizesDropdown.menu?.addItem(NSMenuItem.separator())
+        dimensionRatios.append(1)
+        let commonsizes = [960, 800, 640, 500, 480, 320, 256, 200, 160, 128, 80, 64]
+        for size in commonsizes {
+            let dimensionRatio = CGFloat(size) / dimensions.width
+            dimensionRatios.append(Float(dimensionRatio))
+            let percentageString = "\(Int(round(dimensionRatio * 100.0)))%"
+            predefinedSizesDropdown.addItem(withTitle: "\(Int(dimensions.width * dimensionRatio)) × \(Int(dimensions.height * dimensionRatio)) (\(percentageString))")
+        }
+        if dimensions.width >= 640 {
+            scaleXDoubleValue = 0.5
+            scaleYDoubleValue = 0.5
+            predefinedSizesDropdown.selectItem(at: 3)
+        } else {
+            predefinedSizesDropdown.selectItem(at: 2)
+        }
+        scaleXMinDoubleValue = minimumScale(inputDimensions: dimensions)
+        scaleYMinDoubleValue = scaleXMinDoubleValue
+        updateTextFieldsFromPopup(asPercentage: false)
+        self.recalculateCurrentDimensions()
+    }
+
+    private func updateTextFieldsFromPopup(asPercentage: Bool) {
+        if asPercentage {
+            widthTextField.stringValue = "\(Int(100 * CGFloat(scaleXDoubleValue)))"
+            heightTextField.stringValue = "\(Int(100 * CGFloat(scaleYDoubleValue)))"
+        } else {
+            widthTextField.stringValue = "\(Int(metadata.dimensions.width * CGFloat(scaleXDoubleValue)))"
+            heightTextField.stringValue = "\(Int(metadata.dimensions.height * CGFloat(scaleYDoubleValue)))"
+        }
+    }
+
+    func control(_ control: NSControl, textView: NSTextView, doCommandBy commandSelector: Selector) -> Bool {
```

Comment:
Can you move this logic out into a NSTextField subclass called `IntTextField` that handles the validation and arrow up/down logic? It could also have settings for minimum and maximum value.